ts/queue.ts

```diff
@@ -87,23 +87,33 @@ export default class Queue {
       return;
     }
     this.handler(msg, async (error, reply) => {
-      const { replyTo, correlationId } = msg.properties;
+      const { replyTo, correlationId, headers } = msg.properties;
       if (error && reply !== Queue.STOP_PROPAGATION) {
         reply = Object.assign({}, Queue.ERROR_DURING_REPLY, { error_message: error });
       }
 
       if (!replyTo || reply === Queue.STOP_PROPAGATION) {
         ack();
       } else if (reply instanceof Readable) {
-        const properties = { correlationId, contentType: 'application/json', headers: { isStream: true } };
+        const properties = {
+          correlationId,
+          contentType: 'application/json',
+          headers: { isStream: true, correlationId }
+        };
         try {
+          let id = 0;
           for await (let chunk of reply) {
+            properties.correlationId = `${correlationId}.${id++}`;
             if (chunk instanceof Buffer) chunk = chunk.toString();
-            const replyBuffer = encode(chunk);
-            this.channel.sendToQueue(replyTo, replyBuffer, properties);
+            if (headers.backpressure) {
+              await Queue.getReply(chunk, properties, this.channel, replyTo, null, headers.timeout);
+            } else {
+              this.channel.sendToQueue(replyTo, encode(chunk), properties);
+            }
           }
           this.channel.sendToQueue(replyTo, encode(null), properties, ack);
         } catch (e) {
+          logger.info(`[${correlationId}] -> Publishing to queue ${replyTo} error ${e}`);
```

Comment:
Shouldn't this be logger.error?

Reply:
I was between info and error. Since you also think error is better I will change it.